server/halomod_app/endpoint_models.py

```diff
@@ -102,28 +88,84 @@ def get_models_data_toml():
     # Cache timeout set to 3 seconds, which seems like enough time for the user
     # to change a paremeter and try to download again, but prevents spamming.
     return send_file(buff, as_attachment=True,
-                     attachment_filename="all_plots_toml.zip",
-                     cache_timeout=3)
+                     download_name="all_plots_toml.zip",
+                     max_age=3)
+
+
+@endpoint_models.route('/models', methods=['POST'])
+def create():
+    """Creates multiple models at once and returns the resultant list of names
+    for the session
+    POST /models
+
+    Parameters:
+    ```
+    [
+        {
+            "params": <dictionary_of_params>,
+            "label": <model_name>
+        },
+        {
+            # the same as above for another model, etc.
+        }
+    ]
+    ```
+
+    Returns:
+    - model_names: string[]
+    - example: `{"model_names": <list_of_model_names_in_session>}`
+    """
+    json = request.get_json()
+    if "data" not in json:
+        print('There was no "data" property on the request object')
+    new_models = json['data']
+
+    models = None
+    if 'models' in session:
+        models = pickle.loads(session.get('models'))
+    else:
+        models = {}
+
+    initial_model = get_initial_model()
+
+    # Only allow one model to be created at a time. Prevents crashes in the
+    # server from Fortran when two models are created at the same time.
+    modelCreationSem.acquire()
 
+    num_models = len(models)
+    for model_data in new_models:
+        params = model_data['params']
+        label = model_data['label']
+        models[label] = hmf_driver(
+            previous=initial_model,
+            **params)  # creates model from params
 
-"""Clone model
-PUT /models
+    modelCreationSem.release()
 
-Parameters:
-- model_name: string
-- new_model_name: string
+    print('it got here 2')

```

Comment:
I assume this is meant to be removed :-)

Reply:
Ah yes, good catch! 😂